### ISSUE TITLE
Simplexmlparser and Camlp4

### ISSUE THREAD

**Issue by vasilisp** [ocsigen/tyxml#170]
I am in the process of eliminating some of the antiquated and unnecessary dependencies of Ocsigenserver, direct or not. I would like to get rid of Camlp4. It is currently needed because we use the Camlp4-powered `Simplexmlparser` for parsing configuration files.

A GitHub search reveals quite a few direct users of `Simplexmlparser`.

What would the proper course of action be?

- Use `markup.ml` (or something else?) in Ocsigenserver, and potentially remove the TyXML dependency entirely. Leave `Simplexmlparser` untouched for whoever needs/likes it.
- Implement the `Simplexmlparser` interface with `markup.ml` or otherwise.

In any case, there is a danger of incompatibilities. (I am not sure how standard the XML in `Simplexmlparser` et al. is.)

**Comment by Drup** [comment]
Iirc, `simplexmlparser` in ocsigenserver is not even used to parse html, only config xml files. It's xmlm/xml-light you should use for that.

I don't plan to reimplement `Simplexmlparser`, tyxml is not a parsing library and there are much better alternatives. In fact, I should have deprecated it formally in the last version ...

**Comment by vasilisp** [comment]
Yes, I had the XML part of `markup.ml` in mind. But you are right, `xmlm`/`xml-light` sound more appropriate, with the second being an easier transition due to its common heritage with `Simplexmlparser`. In any case, a migration away from `Simplexmlparser` is imminent.

**Comment by Drup** [comment]
You have all my support to get rid of it :p

**Comment by Drup** [comment]
@vasilisp head's up, I removed the whole camlp4 stuff in the dunification PR, so the xml will be gone by next version. How's the contingency plan going ?
